app/config.go

```diff
@@ -25,38 +27,66 @@ type Config struct {
 	Actions []*action.Config `json:"actions"`
 }
 
-// defaultConfigProvider implementation of ConfigProvider
-type defaultConfigProvider struct {
-}
-
 // ConfigProvider interface to implement to provide the app configuration
 type ConfigProvider interface {
 	GetApp() (*Config, error)
 }
 
-// DefaultSerializer returns the default App Serializer
-func DefaultConfigProvider() ConfigProvider {
-	return &defaultConfigProvider{}
-}
+func LoadConfig(flogoJson string) (*Config, error) {
+	if flogoJson == "" {
+		configPath := config.GetFlogoConfigPath()
+
+		flogo, err := os.Open(configPath)
+		if err != nil {
+			return nil, err
+		}
+
+		file, err := ioutil.ReadAll(flogo)
+		if err != nil {
+			return nil, err
+		}
 
-// GetApp returns the app configuration
-func (d *defaultConfigProvider) GetApp() (*Config, error) {
+		updated, err := AppConfigPreprocess(file)
+		if err != nil {
+			return nil, err
+		}
 
-	configPath := config.GetFlogoConfigPath()
+		app := &Config{}
+		err = json.Unmarshal(updated, &app)
+		if err != nil {
+			return nil, err
+		}
+		return app, nil
+	} else {
+		updated, err := AppConfigPreprocess([]byte(flogoJson))
+		if err != nil {
+			return nil, err
+		}
 
-	flogo, err := os.Open(configPath)
-	if err != nil {
-		return nil, err
+		app := &Config{}
+		err = json.Unmarshal(updated, &app)
+		if err != nil {
+			return nil, err
+		}
+		return app, nil
 	}
+}
+
+func AppConfigPreprocess(appJson []byte) ([]byte, error) {
```

Comment:
I would rename it to preprocessConfig, and leave it private for now until we sort out the interface